### PR TITLE
libvips 8.9.0 compatibility

### PR DESCRIPTION
**Pull request by Elad-Laufer**
Adhere to libvips concurrent metadata update rules

### CODE REVIEW

File: vips/image_test.go [hunk 1]
```diff
@@ -97,8 +97,8 @@ func TestImageRef_Error(t *testing.T) {
 
 	src := bytes.NewReader(srcBytes)
 	img, err := NewImageFromReader(src)
-	assert.Error(t, err)
-	assert.Nil(t, img)
+	assert.NoError(t, err)
```
Comment (golanpa) [inline vips/image_test.go]: why the drastic change in assertion?

File: vips/image_test.go [hunk 2]
```diff
@@ -242,11 +262,36 @@ func TestImageRef_RemoveOrientation__NoEXIF(t *testing.T) {
...
 	defer image.Close()
 
-	image.RemoveOrientation()
+	err = image.RemoveOrientation()
+	require.NoError(t, err)
+
+	assert.Equal(t, 0, image.GetOrientation())
+}
+
+func TestImageRef_RemoveMetadata(t *testing.T) {
```
Comment (golanpa) [inline vips/image_test.go]: please rename and add another part that specifies the expected outcome

File: vips/image_test.go [hunk 3]
```diff
@@ -242,11 +262,36 @@ func TestImageRef_RemoveOrientation__NoEXIF(t *testing.T) {
...
+
+	image, err := NewImageFromFile(resources + "jpg-24bit-icc-adobe-rgb.jpg")
+	require.NoError(t, err)
+	defer image.Close()
 
-	o := image.GetOrientation()
+	err = image.RemoveMetadata()
+	require.NoError(t, err)
+
+	assert.True(t, image.HasICCProfile())
```
Comment (golanpa) [inline vips/image_test.go]: i would also check that metadata was removed

File: vips/image_test.go [hunk 4]
```diff
@@ -242,11 +262,36 @@ func TestImageRef_RemoveOrientation__NoEXIF(t *testing.T) {
...
+}
+
+func TestImageRef_RemoveICCProfile(t *testing.T) {
+	Startup(nil)
+
+	image, err := NewImageFromFile(resources + "jpg-24bit-icc-adobe-rgb.jpg")
+	require.NoError(t, err)
+	defer image.Close()
+
+	err = image.RemoveICCProfile()
```
Comment (golanpa) [inline vips/image_test.go]: also check that metadata not removed?

File: vips/image.go [hunk 5]
```diff
@@ -170,12 +169,28 @@ func (r *ImageRef) GetOrientation() int {
 	return vipsGetMetaOrientation(r.image)
 }
 
-func (r *ImageRef) SetOrientation(orientation int) {
-	vipsSetMetaOrientation(r.image, orientation)
+func (r *ImageRef) SetOrientation(orientation int) error {
+	out, err := vipsCopyImage(r.image)
```
Comment (golanpa) [inline vips/image.go]: we need to understand the penalty of this copy operation. is it a copy-on-write ?

File: vips/image_test.go (hunk 1 above)
Comment (Elad-Laufer) [reply vips/image_test.go]: Because libvips now handles PNG with oversized meta gracefully instead of failing

File: vips/image.go (hunk 5 above)
Comment (Elad-Laufer) [reply vips/image.go]: https://libvips.github.io/libvips/API/current/libvips-conversion.html#vips-copy

> VIPS copies images by copying pointers, so this operation is instant, even for very large images.

https://github.com/lovell/sharp/commit/bb15cd906717ba054d9c23db2971e94d2e250969

**Comment by Elad-Laufer** [comment]
> see questions.
> Also, why the two golden file have changed?

Because they are now half the size

File: vips/image_test.go (hunk 2 above)
Comment (Elad-Laufer) [reply vips/image_test.go]: not a great fan of lengthy func names, especially in simple cases such as this

File: vips/image_test.go (hunk 3 above)
Comment (Elad-Laufer) [reply vips/image_test.go]: done

File: vips/image_test.go (hunk 4 above)
Comment (Elad-Laufer) [reply vips/image_test.go]: done

File: vips/image.go (hunk 5 above)
Comment (golanpa) [reply vips/image.go]: cool. using a pointer

**Pull request merged**